### PR TITLE
drivers: sensor: Add TMP114

### PR DESCRIPTION
**Pull request by fgihl**
Add device driver for TI TMP114 I2C temperature sensor. The driver only support basic functionallity, i.e. fetch temperature using default values.

Datasheet:
	https://www.ti.com/lit/ds/symlink/tmp114.pdf

### CODE REVIEW

File: drivers/sensor/ti/tmp114/tmp114.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#define TMP114_REG_DEVICE_ID	0xb
+
+#define TMP114_RESOLUTION	78125       /* in tens of uCelsius*/
+#define TMP114_RESOLUTION_DIV	10000000
+
+#define TMP114_DEVICE_ID	0x1114
+
+#define TMP114_ALERT_DATA_READY  BIT(0)
+
+struct tmp114_data {
```
Comment (teburd) [inline drivers/sensor/ti/tmp114/tmp114.h:22]: This can all be moved into tmp114.c, no real need for a separate .h here

**Review by teburd — CHANGES_REQUESTED** [review]
Needs to be added to the build_all sensor test https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/drivers/build_all/sensor and I don't think the header is needed (move it all to .c)

Otherwise looks ok to me

**Review by teburd — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -207,6 +207,7 @@ Drivers and Sensors
 * SDHC
 
 * Sensor
+  * Added TMP114 driver
```
Comment (kartben) [inline doc/releases/release-notes-3.7.rst:210]:
```suggestion
* Sensor

  * Added TMP114 driver
```

File: drivers/sensor/ti/tmp114/Kconfig [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# TMP114 temperature sensor configuration options
+
+# Copyright (c) 2024 Fredrik Gihl
+# SPDX-License-Identifier: Apache-2.0
+
+config TMP114
+	bool "TMP114 Temperature Sensors"
```
Comment (kartben) [inline drivers/sensor/ti/tmp114/Kconfig:7]:
```suggestion
	bool "TMP114 Temperature Sensor"
```

**Review by kartben — CHANGES_REQUESTED** [review]
two minor typos including one that breaks the formatting of the release notes

File: drivers/sensor/ti/tmp114/tmp114.c [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+			   enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
```
Comment (MaureenHelm) [inline drivers/sensor/ti/tmp114/tmp114.c:141]: This doesn't do anything, please remove.

File: drivers/sensor/ti/tmp114/tmp114.c [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+};
+
+static int tmp114_init(const struct device *dev)
+{
+	struct tmp114_data *drv_data = dev->data;
+	const struct tmp114_dev_config *cfg = dev->config;
+	int rc;
+	uint16_t id;
+
+	if (!device_is_ready(cfg->bus.bus)) {
```
Comment (MaureenHelm) [inline drivers/sensor/ti/tmp114/tmp114.c:184]: Use `i2c_is_ready_dt`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

**Comment by fgihl** [comment]
Rebased branch and resolve conflict in i2c.dtsi

**Review by teburd — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**